src/cpu/mod.rs

```diff
@@ -628,6 +665,30 @@ impl Cpu {
 
     // gp regs
 
+    pub unsafe fn get_reg64(&self, reg: GpRegs) -> u64 {
```

Comment:
nit: if this is exposed by the rust api it should be reg64()/set_reg64 to be consistent with the rest of the files in this function. get/set are used that the cabi layer